### DISCUSSION TITLE
Fastest way to communicate between cores

### DISCUSSION THREAD

**Discussion by print118** [earlephilhower/arduino-pico#1609] [Q&A]
Hi, would it be faster to communicate between the two cores using memory or FIFO? 

For example, core 1 gets data from sensors and sends it to core 2, which sends the sensor data to serial. Question is, would it be better to use FIFO for sending that sensor data from core 1 to 2, or have core 1 constantly save the data to a variable and have core 2 constantly read the variable to send it to serial. 

Is either one faster or better in some way?

**Comment by earlephilhower** [comment]
Polling would be fastest, of course, but would eat the 2nd core 100% just doing that.  If there's no power constraints and no other use for the core, then I'd say go for it.

OTW you can use the rp2040.fifo object or any of the other multicore comms options to make it happen.  It all depends on what frequency new data comes in and what latency you can support...
